source/module_base/test/matrix3_test.cpp

```diff
@@ -1,8 +1,65 @@
 #include "../matrix3.h"
 #include "gtest/gtest.h"
+#include "gmock/gmock.h"
 #include <random>
 #include <vector>
-class matrix3_test : public testing::Test
+
+/************************************************
+ *  unit test of class Matrix3
+ ***********************************************/
+
+/**
+ * - Tested Functions:
+ *   - Construct
+ *     - two ways of constructing a 3x3 matrix
+ *   - Identity
+ *     - set a 3x3 matrix to identity matrix
+ *   - Zero
+ *     - set all elements of a 3x3 matrix to zero
+ *   - Det
+ *     - calculate the determinant of 3x3 matrix
+ *   - Transpose
+ *     - do the transpose of 3x3 matrix
+ *   - Inverse
+ *     - do the inverse of 3x3 matrix
+ *   - Assignment
+ *     - reloaded Assignment operator "=" for 3x3 matrix
+ *   - AddEqual
+ *     - reloaded operator "+=" for 3x3 matrix
+ *   - MinusEqual
+ *     - reloaded operator "-=" for 3x3 matrix
+ *   - MultiplyEqual
+ *     - reloaded operator "*=" for (3x3 matrix) * number
+ *   - OverEqual
+ *     - reloaded operator "/=" for (3x3 matrix) / number
+ *   - Print
+ *     - print a 3x3 matrix
+ *   - MaddM
+ *     - reloaded operator "+"  for two 3x3 matrix
+ *   - MminusM
+ *     - reloaded operator "-"  for two 3x3 matrix
+ *   - MoverNum
+ *     - reloaded operator "/"  for a (3x3 matrix)/(scalar)
+ *   - MmultiplyM
+ *     - reloaded operator "*"  for (3x3 matrix)*(3x3 matrix)
+ *   - MmultiplyNum
+ *     - reloaded operator "*"  for (3x3 matrix)*(scalar)
+ *     - as well as (scalar)*(3x3 matrix)
```

Comment:
Does "as well as" have the meaning of "similar to" or something like that? I'm not sure about that 😆

Reply:
"as well as" means "and"